front/src/modules/simulationResult/components/SpaceTimeChartWrapper/useLazyProjectTrains.ts

```diff
@@ -45,29 +45,35 @@ const useLazyProjectTrains = ({
   }, []);
 
   useEffect(() => {
-    if (!path) return undefined;
-    const opRefs: OperationalPointReference[] = [];
-    const opDistances: number[] = [];
-    const options = {
+    const baseOptions = {
       dispatch,
       infraId,
-      pathfindingResult: path,
       electricalProfileSetId,
       onProgress,
     };
 
-    if (!operationalPoints || operationalPoints.length === 0) return;
-    operationalPoints.forEach(({ id, position }, index) => {
-      opRefs.push({ operational_point: id });
-      if (index > 0) {
-        opDistances.push(position - operationalPoints[index - 1].position);
-      }
-    });
+    let loader: TrainProjectionLazyLoaderAbstract;
+
+    if (projectionType === 'trackProjection' && path) {
+      loader = new TrainTrackProjectionLazyLoader({
+        ...baseOptions,
+        path,
+      });
+    } else {
+      if (operationalPoints.length < 2) return;
+
+      const opRefs: OperationalPointReference[] = [];
+      const opDistances: number[] = [];
 
-    const loader =
-      projectionType === 'trackProjection'
-        ? new TrainTrackProjectionLazyLoader(options)
-        : new TrainOpProjectionLazyLoader(opRefs, opDistances, options);
+      operationalPoints.forEach(({ id, position }, index) => {
+        opRefs.push({ operational_point: id });
```

Comment:
:warning: Here, you only handle the case where the op are found in the infra : if they're not found, they will be filtered in useProjectPath (and they won't have any operational_point id here)